### ISSUE TITLE
MediaView shows on wrong monitor

### ISSUE THREAD

**Issue by ghost** [telegramdesktop/tdesktop#5126]
### Steps to reproduce
1. You need two monitors. Enable extended desktop mode. Set display 2 as primary monitor.
2. Disconnect monitor 2.
3. Startup Telegram.
4. Connect monitor 2. Windows will automatically move `MainWindow` to monitor 2.
5. Click on a photo or a gif to preview.

### Expected behaviour
`MediaView` shows on monitor 2.

### Actual behaviour
`MediaView` shows on monitor 1.

### Configuration
**Operating system**: Windows 10 1803

**Version of Telegram Desktop**: 1.3.14

**Used theme**: default

**Comment by Alexx-G** [comment]
I have same issue with Telegram Desktop 1.3.14 on Linux (Linux Mint 18.3)

**Comment by smeshny** [comment]
I have same issue with Telegram Desktop 1.3.16 Alpha on Windows 10. To fix this problem u need drag telegram window to second monitor with opened image and drag it back. 
Very annoying issue...

**Comment by poschi3** [comment]
I have the same Issue with Telegram Desktop 1.4.3 and 1.4.4 Beta on Linux Mint 18.3 Cinnamon

**Comment by doubledare704** [comment]
In telegram 1.4.3 with Ubuntu 18.04.1 still appearing to another display even with dragging telegram window to display with image and dragging back

**Comment by poschi3** [comment]
I still have the problem with Telegram Desktop 1.5

**Comment by Cyberuben** [comment]
I have the same issue on Windows with version 1.5.2

**Comment by malucious81** [comment]
I am having  this issue on Fedora 29, gnome desktop. Telegram version 1.5.2

**Comment by kode54** [comment]
> I am having this issue on Fedora 29, gnome desktop. Telegram version 1.5.2
> Edit: This does not occur the first time I open an image, when telegram is freshly launched. Only on the second image and onward.

I can confirm this on Arch. I can also report that it does not occur with Plasma / KWin.

**Comment by ice-bit** [comment]
Same issue on Arch Linux with version 1.5.4(latest version).
As @malucious81 said, this bug appears only the second time you open an image.

**Comment by dadee** [comment]
same issue win7 x64 1.5.4

**Comment by HamishMacEwan** [comment]
Similar issue with Windows 10, Telegram opens on the secondary not primary monitor and its awkward (alt-space, "M," arrow, mouse move) to put it back on the primary where it was during close down.

**Comment by tzurbaev** [comment]
This issue was fixed for me (Ubuntu 18.04; media view windows was displaying on second monitor) after v1.5.11 release. I'm not familiar with the tdesktop's codebase but [this change](https://github.com/telegramdesktop/tdesktop/compare/v1.5.10...v1.5.11#diff-686c1a9e0659ac4315b417a8ceb59e0d) might be the one that fixed this.

**Comment by GoldsteinE** [comment]
Same, Arch Linux + i3wm, Telegram Desktop 1.5.12

**Comment by poschi3** [comment]
I still have the problem with Telegram Desktop 1.5.11 and 1.5.12 on Linux Mint 18.3 Cinnamon

**Comment by malucious81** [comment]
The issue is resolved for me on version 1.5.11 still on Fedora 29 Workstation.

**Comment by kode54** [comment]
> > I am having this issue on Fedora 29, gnome desktop. Telegram version 1.5.2
> > Edit: This does not occur the first time I open an image, when telegram is freshly launched. Only on the second image and onward.
> 
> I can confirm this on Arch. I can also report that it does not occur with Plasma / KWin.

And now I can confirm it's fixed on Gnome, but now broken on Plasma / KWin.

**Comment by john-preston** [comment]
@kode54 .. and that's all you need to know about UI development for Linux 🤣

**Comment by b1oki** [comment]
Has two displays, Primary left №1, second right №2. 
Media popup opens on №2 everytime.

- Linux Mint 19.1 Tessa (based on Ubuntu 18.04) with Cinnamon [_link_](https://linuxmint.com/edition.php?id=261)
- Telegram Desktop 1.6.3-6-g9b247ce (from [snapd](https://snapcraft.io/telegram-desktop))

**Comment by dreamworkerln** [comment]
> Has two displays, Primary left №1, second right №2.
> Media popup opens on №2 everytime.

Same thing in inverse. 
Primary right, secondary left.
Media popup opens on secondary everytime.

**Comment by proea** [comment]
Same issue +1

**Comment by AlexYegupov** [comment]
Same issue 
Ubuntu 18.10
Telegram 1.8.15

**Comment by SDekkers** [comment]
Same issue

Windows 10 (10.0.18362.535)

**Comment by pilot4u** [comment]
same issue on mint 19.2

**Comment by ka4a** [comment]
same issue on linux mint 19.3 with last version of telegram desktop. Please fix this, or provide manual how to fix it.

**Comment by b1oki** [comment]
same issue with some changes from [previous](https://github.com/telegramdesktop/tdesktop/issues/5126#issuecomment-477268099)
```
OS              Linux Mint 19.2
Kernel          5.5.2-050502-generic
Snap package    telegram-desktop 1.9.9-4-ga25792187
```
Two FullHD displays in horizontal, right is primary. Media popup opens on left everytime.

**Comment by w201403** [comment]
Linux Mint 18.3 Cinnamon
Telegram Desktop 1.9.21
Same problem

**Comment by Nazgard** [comment]
Windows 10
Telegram Desktop 2.1.6
Same problem

**Comment by iharuspex** [comment]
Same issue
Arch Linux (kernel 5.7.7-arch1-1)
Telegram Desktop 2.1.13

In my case, the problem appears only with MATE, with Lxqt everything is fine.

**Comment by johnkarpn** [comment]
Same problem

ArchLinux (5.8.5-arch1-1)
Gnome 3.36.5 (Wayland)
Telegram Desktop 2.3.2

Global env:
GDK_BACKEND=x11

telegramdesktop.desktop:
Exec=env QT_QPA_PLATFORM=wayland XCURSOR_SIZE=24 telegram-desktop -- %u

Console output after displaying the image in full-screen mode:
```
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
```

**Comment by joaociocca** [comment]
crap, missed this yesterday to wish it a happy 2-year anniversary! 🎉🎂

**Comment by kemenyadi** [comment]
Windows 10, Telegram v2.3.1.

**Comment by Lungenpest** [comment]
Some Problem here.

Telegram Desktop 2.4.6
Linux Mint 20 Cinnamon

Primary right, secondary left.
Telegram is on the primary.
Media popup opens on secondary every time and **locks** the telegram window.

And when I use "Move to the other monitor" in the taskbar, this setting will be ignored the next time i open the media viewer.

The Telegram main window is locked, when the media viewer is opened, so the media viewer should open on the same monitor like the telegram main window and its better to not lock the main window for better work flow.

![image](https://user-images.githubusercontent.com/64774660/98220795-c2b8e100-1f46-11eb-92b8-0f57a9107381.png)


> System:    Kernel: 5.4.0-52-generic x86_64 bits: 64 compiler: gcc v: 9.3.0 Desktop: Cinnamon 4.6.7 
           wm: muffin dm: LightDM Distro: Linux Mint 20 Ulyana base: Ubuntu 20.04 focal 
Machine:   Type: Desktop Mobo: MSI model: Z77A-G41 (MS-7758) v: 3.0 serial: <filter> 
           BIOS: American Megatrends v: 2.13 date: 03/07/2014 
CPU:       Topology: Quad Core model: Intel Core i5-3570K bits: 64 type: MCP arch: Ivy Bridge 
           rev: 9 L2 cache: 6144 KiB 
           flags: avx lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 27201 
           Speed: 1600 MHz min/max: 1600/6300 MHz Core speeds (MHz): 1: 1600 2: 1600 3: 1600 
           4: 1600 
Graphics:  Device-1: Intel Xeon E3-1200 v2/3rd Gen Core processor Graphics vendor: Micro-Star MSI 
           driver: i915 v: kernel bus ID: 00:02.0 chip ID: 8086:0162 
           Device-2: NVIDIA GP104 [GeForce GTX 1070] vendor: ZOTAC driver: nvidia v: 450.80.02 
           bus ID: 01:00.0 chip ID: 10de:1b81 
           Display: x11 server: X.Org 1.20.8 driver: modesetting,nvidia 
           unloaded: fbdev,nouveau,vesa resolution: 1920x1080~60Hz, 1920x1080~60Hz 
           OpenGL: renderer: GeForce GTX 1070/PCIe/SSE2 v: 4.6.0 NVIDIA 450.80.02 
           direct render: Yes 
Audio:     Device-1: Intel 7 Series/C216 Family High Definition Audio vendor: Micro-Star MSI 
           driver: snd_hda_intel v: kernel bus ID: 00:1b.0 chip ID: 8086:1e20 
           Device-2: NVIDIA GP104 High Definition Audio vendor: ZOTAC driver: snd_hda_intel 
           v: kernel bus ID: 01:00.1 chip ID: 10de:10f0 
           Sound Server: ALSA v: k5.4.0-52-generic 
Network:   Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet vendor: Micro-Star MSI 
           driver: r8169 v: kernel port: d000 bus ID: 03:00.0 chip ID: 10ec:8168 
           IF: enp3s0 state: up speed: 1000 Mbps duplex: full mac: <filter> 
           IF-ID-1: wgpia0 state: unknown speed: N/A duplex: N/A mac: N/A 
Drives:    Local Storage: total: 232.89 GiB used: 37.82 GiB (16.2%) 
           ID-1: /dev/sda vendor: Crucial model: CT250MX500SSD1 size: 232.89 GiB speed: 3.0 Gb/s 
           serial: <filter> 
Partition: ID-1: / size: 227.74 GiB used: 37.82 GiB (16.6%) fs: ext4 dev: /dev/sda5 
Sensors:   System Temperatures: cpu: 37.0 C mobo: N/A gpu: nvidia temp: 59 C 
           Fan Speeds (RPM): N/A gpu: nvidia fan: 0% 
Repos:     No active apt repos in: /etc/apt/sources.list 
           Active apt repos in: /etc/apt/sources.list.d/official-package-repositories.list 
           1: deb http: //packages.linuxmint.com ulyana main upstream import backport #id:linuxmint_main
           2: deb http: //archive.ubuntu.com/ubuntu focal main restricted universe multiverse
           3: deb http: //archive.ubuntu.com/ubuntu focal-updates main restricted universe multiverse
           4: deb http: //archive.ubuntu.com/ubuntu focal-backports main restricted universe multiverse
           5: deb http: //security.ubuntu.com/ubuntu/ focal-security main restricted universe multiverse
           6: deb http: //archive.canonical.com/ubuntu/ focal partner
Info:      Processes: 236 Uptime: 1h 09m Memory: 15.34 GiB used: 3.06 GiB (20.0%) Init: systemd 
           v: 245 runlevel: 5 Compilers: gcc: 9.3.0 alt: 9 Client: Unknown python3.8 client 
           inxi: 3.0.38

**Comment by ilya-fedin** [comment]
The algorithm seems to be valid, I don't understand why this happens at all
https://github.com/telegramdesktop/tdesktop/blob/620639ef83023285e844a4a2305fc1e0634d1a0b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp#L415-L436

**Comment by rusmikev** [comment]
Same problem. 

Telegram Desktop 2.4.7
Linux Mint 20 Cinnamon
rx5500xt with standard mesa driver. What else can I show?

> Some Problem here.
> 
> Telegram Desktop 2.4.6
> Linux Mint 20 Cinnamon
> 
> Primary right, secondary left.
> Telegram is on the primary.
> Media popup opens on secondary every time and **locks** the telegram window.
> 
> And when I use "Move to the other monitor" in the taskbar, this setting will be ignored the next time i open the media viewer.
> 
> The Telegram main window is locked, when the media viewer is opened, so the media viewer should open on the same monitor like the telegram main window and its better to not lock the main window for better work flow.
> 
> ![image](https://user-images.githubusercontent.com/64774660/98220795-c2b8e100-1f46-11eb-92b8-0f57a9107381.png)

**Comment by ilya-fedin** [comment]
> What else can I show?

I have no idea. This seems to be a Qt/Cinnamon/gfx drivers bug, since the tdesktop code just sets the same screen that main window uses.

**Comment by Lungenpest** [comment]
> > What else can I show?
> 
> I have no idea. This seems to be a Qt/Cinnamon/gfx drivers bug, since the tdesktop code just sets the same screen that main window uses.

Ok, and now? Who should we contact? A solution would be great, because this bug is very annoying in my daily work...

**Comment by ilya-fedin** [comment]
> And I see some people here have this problem also in Windows 10.

As far as I'm understanding, there mixed three issues:
1. Qt doesn't handle the screen change when a screen is connected on Windows and Windows automatically moves the window to that screen, so that it is necessary to move window to another screen and back (there were multiple Qt updates, the last one is at 2.4.8, so maybe not an issue anymore)
2. Wayland doesn't have a protocol to specify window position, so that media viewer is appearing on the screen compositor is chosen and not tdesktop, that's the cause on GNOME Wayland. Unfortunately, no way to fix since there's just no such protocol.
3. Some weird issue on Cinnnamon, I can't reproduce that on a VM, media viewer successfully open on the right screen.

**Comment by ilya-fedin** [comment]
> Who should we contact?

Qt or Cinnamon developers. I don't know on which side the bug so I don't who you should contact.

**Comment by hanthecat** [comment]
Still having this issue 2 years later. Telegram windows desktop v2.47. Would it be possible to add something to the right click context menu and ask the user where they want the window to open?

**Comment by ilya-fedin** [comment]
@hanthecat this issue is about media viewer, not about context menu

**Comment by snitko** [comment]
Tip for those using Telegram Desktop on i3wm:

1. Open the media. If it opens on the wrong dislpay, that window is still going to be in focus.
2. Hit the key combination to move it to the scratchpad.
3. Move your focus to the display you'd like it show on.
3. Hit the key combination to bring it out of the scratchpad and it will show as a floating window on your current display.

**Comment by amatas** [comment]
Could it be possible to change the behavior of the media viewer to use the full window of Telegram as view space instead of the full screen?. This could solve this issue, and also it would be less invasive at any desktop interface. (I don't like to hide other windows if I'm watching a media in Telegram)

**Comment by ilya-fedin** [comment]
@amatas #1710

**Comment by Yakoot** [comment]
same problem

OS: Manjaro 20.2 Nibia
Kernel: x86_64 Linux 5.9.10-1-MANJARO
DE: KDE 5.76.0 / Plasma 5.20.4
WM: KWin
GTK Theme: Breath [GTK2/3]
Icon Theme: breath2

**Comment by ilya-fedin** [comment]
@Yakoot X11 or Wayland?

**Comment by Yakoot** [comment]
> @Yakoot X11 or Wayland?

x11

**Comment by ilya-fedin** [comment]
@Yakoot can't reproduce, opens on the right screen in my VM with KDE Neon

**Comment by lilydjwg** [comment]
Starting with version 2.5.1 this happens to me as well. Arch Linux + X11 + Awesome 3.5.9.

The media viewer is created on the right, primary monitor (same as the Telegram window) but then moved to the left one quickly (I can see a flash on the primary monitor). When I try to move it to the primary monitor, it moves back.

After a restart, it behaves as expected until some unknown condition happens.

**Comment by ilya-fedin** [comment]
someone please fix all these glitchy window managers

**Comment by koos147** [comment]
i had a surface book. no problems. (intel and NVidia graphics )
now i have a surface laptop 2. (only intel graphics) and the media opens on screen 1 (laptop screen) while TG is on the external monitor (2 or 3)
desktop (asus mobo with NVidia graphics) no problem with 3 screens)

**Comment by BennyBeat** [comment]
Same issue on Ubuntu 20.04 and Telegram Beta/Stable. version on Asus Desktop motherboard + GeForce GTX1060. Tested first using two 27" monitors and later using same two monitors + one 75" Smart media LCD TV. No matter if I'm using two or three monitors/TV; problem persist...

Solved switching one of them (from left to right or vice-versa to adjust media Viewer to be showed on correct screen (normally the right hand in my case). This little trick solves also the trouble on Windows 10 (I have dual OS on my computer). Hope this trick can help any other Telegram users to solve the issue (but great if TG can detect right combination or add an option to drag & drop the Media Viewer).

**Comment by ilya-fedin** [comment]
> but great if TG can detect right combination

TG gets all it can from Qt, if one know better APIs, there are the pull requests tab

**Comment by bakatrouble** [comment]
![image](https://user-images.githubusercontent.com/7340354/104105380-b2164580-52be-11eb-988e-cc8325399405.png)
On Windows, MediaView opens with incorrect window size (screen height is 1440px), breaks for both photos and videos only after opening a video.

**Comment by ilya-fedin** [comment]
@bakatrouble 10084

**Comment by stachenov** [comment]
Windows 10 20H2, v2.5.1. A 24" display (primary, left) and a 55" TV (secondary, right). Not only media viewer opens on the TV, the app itself also insists on doing the same on start. Was a big problem with my previous keyboard that lacked the Win key because the TV is normally switched off, but the video card doesn't know that, so it remains active, and I couldn't see the app or move it back. Now I just Shift+Win+Left it every time I start it or open media viewer. But it's still annoying.

**Comment by ilya-fedin** [comment]
@stachenov this issue is about media viewer only

**Comment by sadransh** [comment]
Same Issue. 
Any temp workaround? 

OS: Ubuntu 20.04 -GNOME 3 
Kernel: 5.4.0-62
DE: KDE 5.76.0 / Plasma 5.20.4
graphics : nvidia gtx 1080 . driver version 460

**Comment by Yakoot** [comment]
As for me it shows on wrong monitor in not usual case:
1) Turn on the laptop and launch telegram
2) Connect second display which turns into primary display, so telegram moves there
3) Open mediaview, it shows on laptop display instead of second one

Workaround to fix this problem: I have to move telegram to laptop and then back to external monitor

**Comment by ilya-fedin** [comment]
> As for me it shows on wrong monitor in not usual case:

These reproduction steps are in the first post...

**Comment by Yakoot** [comment]
Oops, sorry =)

**Comment by DASPRiD** [comment]
> Same Issue.
> Any temp workaround?

I found a "fix", although I don't know how permanent that one is. I opened a MediaView, opened the Overview Mode of Gnome (Super-Key) and dragged the MediaView onto the correct monitor. After closing the MediaView and opening it again with another photo, it displayed on the correct monitor.

I haven't verified yet whether Telegram remembers the placement after a restart/reboot.

**Comment by yarreg** [comment]
@DASPRiD Same workaround for Windows.

**Comment by sadransh** [comment]
@DASPRiD for me it did not remember even after closing and opening the telegram. However, it is working Thanks.

**Comment by ilya-fedin** [comment]
To all:
Some debug logging was added to 2.5.8 for media viewer. Debug logs could be activated by `debugmode` cheat code in settings. When you open media viewer, some lines with `Viewer Pos:` should be written to debug logs. Debug logs should be in `DebugLogs` subfolder of profile folder (profile folder is usually `~/.local/share/TelegramDesktop` on Linux, but may differ)

**Comment by sadransh** [comment]
@ilya-fedin 

[17:45:44.146 00-0005047] Viewer Pos: Shown, screen number: 1 (media_view_overlay_widget.cpp : 1328)
[17:45:44.146 00-0005048] Viewer Pos: Currently on screen 1, moving to screen 0 (media_view_overlay_widget.cpp : 450)
[17:45:44.146 00-0005049] Viewer Pos: Screen changed to: 0 (media_view_overlay_widget.cpp : 1341)
[17:45:44.146 00-0005050] Viewer Pos: New actual screen: 0 (media_view_overlay_widget.cpp : 455)
[17:45:44.159 00-0005051] Viewer Pos: Screen changed to: 1 (media_view_overlay_widget.cpp : 1341)
[17:45:44.159 00-0005052] Viewer Pos: Currently on screen 1, moving to screen 0 (media_view_overlay_widget.cpp : 450)
[17:45:44.159 00-0005053] Viewer Pos: Screen changed to: 0 (media_view_overlay_widget.cpp : 1341)
[17:45:44.159 00-0005054] Viewer Pos: New actual screen: 0 (media_view_overlay_widget.cpp : 455)
This is my log from ubuntu 18. it opens on a wrong monitor.

**Comment by ilya-fedin** [comment]
@sadransh how long the bug is happening for you?

**Comment by sadransh** [comment]
@ilya-fedin  
If I understand you correctly, it's a few weeks(since I started using >1 monitor)  I am dealing with this. 
for this log, I just clicked on an image and closed it. I tested this again a few moments ago and the log is almost the same. the image directly opens in the monitor connected with HDMI, I don't see the viewer in other monitors(not even for a moment) ( unless I move it manually)

**Comment by DASPRiD** [comment]
> @DASPRiD for me it did not remember even after closing and opening the telegram. However, it is working Thanks.

Can confirm, the workaround doesn't survive a restart of Telegram.

> @sadransh how long the bug is happening for you?
2.5.7 for me, so quite recent.

**Comment by stachenov** [comment]
I have 2.5.7 now, and the bug seems to be gone. Both bugs, actually. Both the app itself and the media view open up on my primary display now.

**Comment by iqfareez** [comment]
2.5.8. Seems like this issue is gone now. Image/videos now opening on the same screen as telegram.

**Comment by DASPRiD** [comment]
Cannot confirm – did fixing it on Windows break it on Linux?

**Comment by yarreg** [comment]
For me on Windows problem is not resolved.

**Comment by ilya-fedin** [comment]
> did fixing it on Windows break it on Linux?

Attempt to fix #7724 on Linux broke it on Linux

**Comment by sh4dowb** [comment]
> > Same Issue.
> > Any temp workaround?
> 
> I found a "fix", although I don't know how permanent that one is. I opened a MediaView, opened the Overview Mode of Gnome (Super-Key) and dragged the MediaView onto the correct monitor. After closing the MediaView and opening it again with another photo, it displayed on the correct monitor.
> 
> I haven't verified yet whether Telegram remembers the placement after a restart/reboot.

thank you! I was going crazy lol. this works under Ubuntu 20.04 LTS everyone

**Comment by DASPRiD** [comment]
Here is a semi-permanent fix, which will work across Telegram restarts (for Linux only I think): Install `devilspie` from your repository (devilspie 2 should work as well, but has a different configuration language) and set up a config for it, e.g. under `~/.devilspie/telegram-mediaviewer.ds`:

```
(if
  (is (window_name) "Media viewer")
  (begin
    (geometry "1920x1080+5120+0")
  )
)
```

Adjust the geometry to match your screen you want the media viewer on (first portion is the size, second are the coordinates of the upper left corner). Then put devilspie in your startup programs of your DM session. For the current session, just run devilspie in a terminal.

The first time you open a media viewer, it will briefly flash on the primary screen, but then quickly move over to the correct one and stay there until you restart Telegram.

**Comment by SamSpiri** [comment]
2.5.8 , ubuntu 20.04 gnome
2 screens + laptop screen. media viewver is always on laptop screen. I use @DASPRiD fix - press super key then drug media viewer to the screen I want.

**Comment by Paramood** [comment]
Got the same issue with a recent Telegram update on a laptop with Ubuntu. Also, I have 2 external monitors connected.

### Actual behavior:
All images/videos are always opened on the same monitor and it doesn't matter on which monitor Telegram itself is opened. Also, it doesn't matter whether this monitor is set as a primary or a secondary in Ubuntu settings. Seems that Telegram chooses left upper monitor to open media with some exceptions. I'm adding cases with 2 and 3 monitors. I also noticed that Telegram changed its logic on choosing a monitor after several display reconfigurations.

**Noticed cases with different display config in chronological order (arrow shows the monitor on which Telegram opens media):**
![image](https://user-images.githubusercontent.com/26360776/107950380-4bf89e80-6f9f-11eb-93e4-0a8677bd1eeb.png)
![image](https://user-images.githubusercontent.com/26360776/107950401-5450d980-6f9f-11eb-87cd-34a1eb374f36.png)
![image](https://user-images.githubusercontent.com/26360776/107950421-5c107e00-6f9f-11eb-8582-34a278891a25.png)
![image](https://user-images.githubusercontent.com/26360776/107953254-91b76600-6fa3-11eb-906c-346e8696af20.png)
![image](https://user-images.githubusercontent.com/26360776/107953255-91b76600-6fa3-11eb-97c9-19141c7cc8c5.png)
![image](https://user-images.githubusercontent.com/26360776/107953253-91b76600-6fa3-11eb-9218-60c9b4f165af.png)
![image](https://user-images.githubusercontent.com/26360776/107953256-91b76600-6fa3-11eb-886d-954076367ef5.png)
![image](https://user-images.githubusercontent.com/26360776/107953257-91b76600-6fa3-11eb-8413-86fe71ecff01.png)
![image](https://user-images.githubusercontent.com/26360776/107953258-91b76600-6fa3-11eb-8d5f-936c4c3a4071.png)
![image](https://user-images.githubusercontent.com/26360776/107953260-9419c000-6fa3-11eb-80ad-5682d6ac9fab.png)

### Expected behavior:
All media should be opened on the same monitor where Telegram app is opened (just as it was before).

### Configuration:
**Telegram version:** 2.5.8 via snap package manager
**OS:** Ubuntu 20.04.2

**Comment by genetus** [comment]
Add additional info. Bug can be reproduce only on wayland server. On X server all - ok.

**Comment by ilya-fedin** [comment]
> Bug can be reproduce only on wayland server.

There are multiple bugs :see_no_evil: . The wayland one is unfixable since there are just no support for setting position of a window on Wayland.

**Comment by Paramood** [comment]
For me this case is fixed now in Telegram 2.5.9 on Ubuntu 20.04

**Comment by sadransh** [comment]
I am on 2.6.1 on Ubuntu 20.04 and it is fixed. Not sure starting with which update
